core/domain/suggestion_jobs_one_off.py

```diff
@@ -179,3 +179,54 @@ def reduce(key, values):
                 no_of_suggestions_migrated)])
         else:
             yield (key, values)
+
+
+class PopulateSuggestionLanguageCodeMigrationOneOffJob(
+        jobs.BaseMapReduceOneOffJobManager):
+    """A reusable one-time job that may be used to add the language_code field
+    to suggestions that do not have that field yet. The language_code field
+    allows question and translation suggestions to be queried by language.
+    This job will load all existing suggestions from the data store, update
+    them, if needed, and immediately store them back into the data store.
+    """
+
+    _VALIDATION_ERROR_KEY = 'validation_error'
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [suggestion_models.GeneralSuggestionModel]
+
+    @staticmethod
+    def map(item):
+        # Exit early if the suggestion has been marked deleted, or if the
+        # suggestion has already set the language code property, or if the
+        # suggestion type is not queryable by language, since ndb automatically
+        # sets properties that aren't intialized to None.
+        if item.deleted or item.language_code or item.suggestion_type == (
+                suggestion_models.SUGGESTION_TYPE_EDIT_STATE_CONTENT):
+            return
+
+        # Getting the suggestion from the model creates the corresponding
+        # suggestion domain object. In the init functions for the suggestion
+        # domain classes, the language code field is retrieved and set
+        # automatically.
+        suggestion = suggestion_services.get_suggestion_from_model(item)
+        try:
+            suggestion.validate()
+        except Exception as e:
+            logging.error(
+                'Suggestion %s failed validation: %s' % (
+                    item.id, e))
+            yield (
+                PopulateSuggestionLanguageCodeMigrationOneOffJob
+                ._VALIDATION_ERROR_KEY,
+                'Suggestion %s failed validation: %s' % (
+                    item.id, e))
+            return
+        item.language_code = suggestion.language_code
```

Comment:
I was expecting `item.language_code` and `suggestion.language_code` to be the same as the domain object is created out of the model object. Assume in future there's a model with langauge_code None for translation suggestion. So if we call the validation function on the object created using that model, are we expecting the validation check to fail?

Reply:
item.language_code is set to the suggestion.language_code, suggestion.language_code is the language code of the domain object. I'm confused - is there an issue with line 226? The validation check would fail if a translation suggestion's language code was None - I added this check in after @nithusha21 's comments on this PR.